modules/twinklearv.js

```diff
@@ -524,7 +524,7 @@ Twinkle.arv.callback.evaluate = function(e) {
 				reason += ' ' + types;
 			}
 			if (comment !== '') {
-				reason += (reason === '' ? '' : '. ') + comment;
+				reason += (/([.?!;:]|^)$/.test(reason) ? '' : '.') + (reason === '' ? '' : ' ') + comment; // Ends sentence if needed, does nothing if empty string
```

Comment:
Two ternaries in the same sentence isn't super readable. Maybe extracting the ternaries here will make it more readable and self-documenting? Probably wouldn't need the comment after refactoring.

```js
var reasonNeedsPunctuation = /([.?!;:]|^)$/.test(reason);
reason += (reasonNeedsPunctuation ? '' : '.');
var reasonIsBlank = reason === '';
reason += (reasonIsBlank ? '' : ' ');
reason += comment;
```